crates/client/rpc/src/versions/v0_8_0/get_storage_proof.rs

```diff
@@ -0,0 +1,185 @@
+use super::{
+    ContractLeavesDataItem, ContractStorageKeysItem, ContractsProof, GetStorageProofResult, GlobalRoots, MerkleNode,
+    NodeHashToNodeMappingItem,
+};
+use crate::{
+    errors::{StarknetRpcApiError, StorageProofLimit, StorageProofTrie},
+    utils::ResultExt,
+    Starknet,
+};
+use bitvec::{array::BitArray, order::Msb0, slice::BitSlice};
+use jsonrpsee::core::RpcResult;
+use mc_db::{bonsai_identifier, db_block_id::DbBlockId, BasicId, GlobalTrie};
+use starknet_core::types::{BlockId, BlockTag, Felt};
+use starknet_types_core::hash::StarkHash;
+
+fn path_to_felt(path: &BitSlice<u8, Msb0>) -> Felt {
+    let mut arr = [0u8; 32];
+    let slice = &mut BitSlice::from_slice_mut(&mut arr)[5..];
+    slice[..path.len()].copy_from_bitslice(path);
+    Felt::from_bytes_be(&arr)
+}
+
+/// Returns (root hash, nodes)
+fn make_trie_proof<H: StarkHash + Send + Sync>(
+    block_n: u64,
+    trie: &mut GlobalTrie<H>,
+    _trie_name: StorageProofTrie,
+    identifier: &[u8],
+    keys: Vec<Felt>,
+) -> RpcResult<(Felt, Vec<NodeHashToNodeMappingItem>)> {
+    let mut keys: Vec<_> = keys.into_iter().map(|f| BitArray::new(f.to_bytes_be())).collect();
+    keys.sort();
+
+    let mut storage = trie
+        .get_transactional_state(BasicId::new(block_n), trie.get_config())
+        .map_err(|err| anyhow::anyhow!("{err:#}"))
+        .or_internal_server_error("Getting transactional state")?
+        .ok_or(StarknetRpcApiError::CannotMakeProofOnOldBlock)?;
+
+    let root_hash = storage
+        .root_hash(identifier)
+        .map_err(|err| anyhow::anyhow!("{err:#}"))
+        .or_internal_server_error("Getting root hash of trie")?;
+
+    let proof = storage
+        .get_multi_proof(identifier, keys.iter().map(|k| &k.as_bitslice()[5..]))
+        .map_err(|err| anyhow::anyhow!("{err:#}"))
+        .or_internal_server_error("Error while making storage multiproof")?;
+
+    Ok((
+        root_hash,
+        proof
+            .0
+            .into_iter()
+            .map(|(node_hash, n)| NodeHashToNodeMappingItem {
+                node_hash,
+                node: match n {
+                    mc_db::ProofNode::Binary { left, right } => MerkleNode::Binary { left, right },
+                    mc_db::ProofNode::Edge { child, path } => {
+                        MerkleNode::Edge { child, path: path_to_felt(&path), length: path.len() }
+                    }
+                },
+            })
+            .collect(),
```

Comment:
Can you extract this out into a variable please. It's doing too much to be inlined.

Reply:
sure